pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -288,6 +297,66 @@ func (cloud *CloudProvider) insertRegionalDisk(ctx context.Context, volKey *meta
 	return nil
 }
 
+func (cloud *CloudProvider) insertRegionalAlphaDisk(ctx context.Context, volKey *meta.Key, diskType string, capBytes int64, capacityRange *csi.CapacityRange, replicaZones []string, snapshotID, diskEncryptionKmsKey string, multiWriter bool) error {
+	diskToCreateAlpha := &computealpha.Disk{
```

Comment:
Thanks for doing this, It looks pretty good as-is but I think there's also a way to de-dupe some of this copy-pasted code.

Assuming zonal here:
1) Have one `insertDisk` that takes in all the parameters
2) Create a v1 disk with logic up to line `318` here.
3) Make `insertOp` a `interface{}` type (this part is kind of nasty, suggestions welcome)
3) ```
if multiWriter { 
  convertV1DiskToV1AlphaDisk(disk) // you have to write this
  insertOp = cloud.alphaService.Insert(disk)
} else{
  insertOp = cloud.service.Insert(dis)
}
```
Then the rest of the error handling logic is shared
4) typeAssert on `insertOp` and call the respective `waitForOp` on it
5) Share the other logic

This would dedupe most of the code and only have branching for the actual GCE API Insert call and the WaitForOp.
Let me know what you think. Happy to discuss more

Reply:
Thanks for the review. I notice the `waitForOp` methods only refer to `op.Name`. I wonder if it's okay to call the non-alpha operations API to check on the status of an alpha operation. My thinking there is that we're not leveraging alpha features of the operations API.
I'm not sure so I'll err on the side of caution here. The solution you recommended works just fine.

Reply:
ack. I didn't notice that - we could just grab the op name then and avoid the whole type assertion thing

Reply:
wait... looked at it again and we still might need to since we're doing the op get on `svc.ZoneOperations.Get(project, zone, op.Name).Context(ctx).Do()` - thats the v1 service. If we have an alpha service op maybe we can't "find it" unless we use a `v1alpha.ZoneOperations` call? Could you verify

Reply:
David, I ran some tests and found that the v1 operations API is able to get the status of beta and alpha operations.

This can easily be tested using the in-browser API Explorer:
https://cloud.google.com/compute/docs/reference/rest/beta/disks/insert
https://cloud.google.com/compute/docs/reference/rest/v1/zoneOperations/get

I think there also might be some opportunity to reduce the duplication between Zonal and Regional methods but I'll skip that for now. You can let me know your thoughts there. I'm happy to make additional changes.

In the meantime I'll address the rest of your comments and add another commit.

Reply:
lets keep this change focused on adding the multiwriter stuff - feel free to open up a fix to reduce zonal+regional duplication afterwards if you're interested. That would be really cool :+1: 

Ack on the `ops` - lets dedupe them and make sure to add a comment in the code with that exact finding (so someone doesn't come in and waste cycles trying to "fix" it later)